tests/end-to-end/external-solution-integration/internal/scenario/compass_e2e.go

```diff
@@ -0,0 +1,163 @@
+package scenario
+
+import (
+	"crypto/tls"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/vrischmann/envconfig"
+
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/helpers"
+
+	"github.com/kyma-project/kyma/common/resilient"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/internal"
+
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/step"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testkit"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testsuite"
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/rest"
+)
+
+// CompassE2E executes complete external solution integration test scenario
+// using Compass for Application registration and connectivity
+type CompassE2E struct {
+	domain            string
+	testID            string
+	skipSSLVerify     bool
+	applicationTenant string
+	applicationGroup  string
+	lambdaPort        int
+}
+
+// AddFlags adds CLI flags to given FlagSet
+func (s *CompassE2E) AddFlags(set *pflag.FlagSet) {
+	pflag.StringVar(&s.domain, "domain", "kyma.local", "domain")
+	pflag.StringVar(&s.testID, "testID", "compass-e2e-test", "domain")
+	pflag.BoolVar(&s.skipSSLVerify, "skipSSLVerify", false, "Skip verification of service SSL certificates")
+	pflag.StringVar(&s.applicationTenant, "applicationTenant", "", "Application CR Tenant")
+	pflag.StringVar(&s.applicationGroup, "applicationGroup", "", "Application CR Group")
+	pflag.IntVar(&s.lambdaPort, "lambdaPort", 8080, "Lambda port")
+}
+
+// Steps return scenario steps
+func (s *CompassE2E) Steps(config *rest.Config) ([]step.Step, error) {
+	state, err := s.NewState()
+	if err != nil {
+		return nil, err
+	}
+
+	kymaClients := testkit.InitClients(config, s.testID)
+	compassConnector := testkit.NewCompassConnectorClient(s.skipSSLVerify)
```

Comment:
Why not implement this the way `kymaClients` are done. Right now testkit holds the function that is called InitClients which can be renamed to InitKymaClients and you can add InitCompassClients that will hold the connector and director client initialization. That way you will have a consistent way of clients initialization.